### PR TITLE
put store to PD after bootstrap finish

### LINKED ISSUE #5269: put store to PD to early when start
图片
图片

### PR DESCRIPTION
**Pull request by hicqu**
Signed-off-by: qupeng <qupeng@pingcap.com>

## What have you changed? (mandatory)
Call PD put_store API after the store is bootstrapped.
Close #5269 

## What are the type of changes? (mandatory)
Improvement.

## How has this PR been tested? (mandatory)
CI.

### CODE REVIEW

File: src/raftstore/store/fsm/store.rs [hunk 1]
```diff
@@ -1060,15 +1058,16 @@ impl RaftBatchSystem {
             mailboxes.push((fsm.region_id(), BasicMailbox::new(tx, fsm)));
         }
         self.router.register_all(mailboxes);
+
         // Make sure Msg::Start is the first message each FSM received.
+        for addr in address {
```
Comment (BusyJay) [inline src/raftstore/store/fsm/store.rs]: @5kbpers Is the adjustment OK? I recall that you have also changed the boot order of batch system.

File: src/server/node.rs [hunk 2]
```diff
@@ -159,6 +156,11 @@ where
             coprocessor_host,
             importer,
         )?;
+
+        // Put store only if the cluster is bootstrapped.
+        info!("put store to PD"; "store" => ?&self.store);
```
Comment (BusyJay) [inline src/server/node.rs]: @overvenus PTAL, the bootstrap and startup procedure is updated.

File: src/server/server.rs [hunk 3]
```diff
@@ -184,7 +184,9 @@ impl<T: RaftStoreRouter, S: StoreAddrResolver + 'static> Server<T, S> {
             Arc::clone(&cfg),
         );
         box_try!(self.snap_worker.start(snap_runner));
+
         let mut grpc_server = self.builder_or_server.take().unwrap().right().unwrap();
+        info!("listening on addr"; "addr" => &self.local_addr);
```
Comment (overvenus) [inline src/server/server.rs]: Dup with L170

File: src/raftstore/store/fsm/store.rs (hunk 1 above)
Comment (5kbpers) [reply src/raftstore/store/fsm/store.rs]: LGTM

**Pull request merged**